tuplex/test/io/OrcWriteTest.cc

```diff
@@ -0,0 +1,373 @@
+    using namespace tuplex;
+    ContextOptions co = ContextOptions::defaults();
+    Context context(co);
+    auto& ds = context.parallelize(rows);
+    if (!columnNames.empty()) {
+        ds.setColumns(columnNames);
+    }
+    writeDataSet(ds);
+}
+
+TEST(ORC, RowToStr) {
+    using namespace tuplex;
+    std::vector<std::string> cols({"one", "two", "three"});
+
+    auto row1 = Row(1, 2, 3);
+    EXPECT_EQ(R"({"": 1, "": 2, "": 3})", rowToORCString(row1));
+    EXPECT_EQ(R"({"one": 1, "two": 2, "three": 3})", rowToORCString(row1, cols));
+
+    auto row2 = Row(1.1, 2.2, 3.3);
+    EXPECT_EQ(R"({"": 1.1, "": 2.2, "": 3.3})", rowToORCString(row2));
+    EXPECT_EQ(R"({"one": 1.1, "two": 2.2, "three": 3.3})", rowToORCString(row2, cols));
+
+    auto row3 = Row(true, true, false);
+    EXPECT_EQ(R"({"": true, "": true, "": false})", rowToORCString(row3));
+    EXPECT_EQ(R"({"one": true, "two": true, "three": false})", rowToORCString(row3, cols));
+
+    auto row4 = Row("abc", "de", "f");
+    EXPECT_EQ(R"({"": "abc", "": "de", "": "f"})", rowToORCString(row4));
+    EXPECT_EQ(R"({"one": "abc", "two": "de", "three": "f"})", rowToORCString(row4, cols));
+
+    auto row5 = Row(Tuple(1, Tuple(2, 3)), Tuple(Tuple(4, 5), 6, 7, Tuple(8,9)), 10);
+    EXPECT_EQ(R"({"": {"": 1, "": {"": 2, "": 3}}, "": {"": {"": 4, "": 5}, "": 6, "": 7, "": {"": 8, "": 9}}, "": 10})", rowToORCString(row5));
+    EXPECT_EQ(R"({"one": {"": 1, "": {"": 2, "": 3}}, "two": {"": {"": 4, "": 5}, "": 6, "": 7, "": {"": 8, "": 9}}, "three": 10})", rowToORCString(row5, cols));
+
+    auto row6 = Row(List(1,2,3), List(1.1, 2.2, 3.3), List(true));
+    EXPECT_EQ(R"({"": [1, 2, 3], "": [1.1, 2.2, 3.3], "": [true]})", rowToORCString(row6));
+
+    std::string i64_to_f64 = std::string(R"({"1":1.1,"2":2.2,"3":3.3})");
+    std::string f64_to_i64 = std::string(R"({"1.1":1,"2.2":2,"3.3":3})");
+    std::string str_to_bool = std::string(R"({"a":true,"b":false,"c":true})");
+    std::string bool_to_str = std::string(R"({"true":"a","false":"b","true":"c"})");
+    auto row7 = Row(Field::from_str_data(i64_to_f64, python::Type::makeDictionaryType(python::Type::I64, python::Type::F64)),
+                    Field::from_str_data(f64_to_i64, python::Type::makeDictionaryType(python::Type::F64, python::Type::I64)),
+                    Field::from_str_data(str_to_bool, python::Type::makeDictionaryType(python::Type::STRING, python::Type::BOOLEAN)),
+                    Field::from_str_data(bool_to_str, python::Type::makeDictionaryType(python::Type::BOOLEAN, python::Type::STRING)));
+    EXPECT_EQ(R"({"": [{"key": 1, "value": 1.1}, {"key": 2, "value": 2.2}, {"key": 3, "value": 3.3}], "": [{"key": 1.1, "value": 1}, {"key": 2.2, "value": 2}, {"key": 3.3, "value": 3}], "": [{"key": "a", "value": true}, {"key": "b", "value": false}, {"key": "c", "value": true}], "": [{"key": true, "value": "a"}, {"key": false, "value": "b"}, {"key": true, "value": "c"}]})", rowToORCString(row7));
+
```

Comment:
Do we have some tests where an orc file is written, then read and it's checked contents are the same?